analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Binary.cs

```diff
@@ -881,4 +881,71 @@ public void Binary_BitAnd_Range(string expression, int? expectedMin, int? expect
         var validator = SETestContext.CreateCS(code, "int i, int j").Validator;
         validator.TagValue("Value").Should().HaveOnlyConstraints(ObjectConstraint.NotNull, NumberConstraint.From(expectedMin, expectedMax));
     }
+
+    [DataTestMethod]
+    [DataRow(0b0000, 0b0000, 0b0000)]
+    [DataRow(0b0101, 0b0101, 0b0101)]
+    [DataRow(0b0101, 0b0001, 0b0101)]
+    [DataRow(0b1010, 0b0110, 0b1110)]
+    [DataRow(0b1010, 0b0000, 0b1010)]
+    [DataRow(0b1111, 0b1111, 0b1111)]
+    [DataRow(5, -5, -1)]
+    [DataRow(5, -4, -3)]
+    [DataRow(-5, -5, -5)]
+    [DataRow(-5, -4, -1)]
+    public void Binary_BitOr_SingleValue(int left, int right, int expected)
+    {
+        var code = $"""
+            var left = {left};
+            var right = {right};
+            var value = left | right;
+            Tag("Value", value);
+            """;
+        SETestContext.CreateCS(code).Validator.TagValue("Value").Should().HaveOnlyConstraints(ObjectConstraint.NotNull, NumberConstraint.From(expected));
+    }
+
+    [DataTestMethod]
+    [DataRow("i >=  3 && j >=  5", 5, null)]
+    [DataRow("i >=  3 && j >= -5", -5, null)]
+    [DataRow("i >=  3 && j <=  5", null, null)]
+    [DataRow("i >=  3 && j <= -5", null, -1)]
+    [DataRow("i >= -3 && j >=  5", -3, null)]
+    [DataRow("i >= -3 && j >= -5", -5, null)]
+    [DataRow("i >= -3 && j <=  5", null, null)]
+    [DataRow("i >= -3 && j <= -5", null, -1)]
+    [DataRow("i <=  3 && j >=  5", null, null)]
+    [DataRow("i <=  3 && j <=  5", null, 7)]
+    [DataRow("i <=  3 && j >= -5", null, null)]
+    [DataRow("i <=  3 && j <= -5", null, -1)]
+    [DataRow("i <= -3 && j >=  5", null, -1)]
+    [DataRow("i <= -3 && j <=  5", null, -1)]
+    [DataRow("i <= -3 && j >= -5", null, -1)]
+    [DataRow("i <= -3 && j <= -5", null, -1)]
+    [DataRow("i ==  3 && j >=  5", 5, null)]  // (i | j) >= 7, we only get a lower bound of min
+    [DataRow("i ==  3 && j <= -5", null, -1)] // (i | j) <= -5, we only get an upper bound of max
```

Comment:
In hindsight, I picked terrible numbers with 3 and 5. Please find better numbers. Maybe 10 and 16 could work?

Reply:
What we need for `|` is a couple of numbers a and b such that:
- a >= 0 and b >= 0
- `[0, a] | [0, b]` contains a number > a | b

3 and 5 are not good, because `0b00000011 | 0b00000101 = 0b00000111`, which is the max of `[0, 3] | [0, 5]`.

I think 4 and 6 are good enough (and single digits): `(0b00000100 | 0b00000110) = 0b00001100 = 6 < 0b00000111 = 7`, where you can get `0b00000111 = 7` by `0b00000011 = 3`, which belongs to `[0, 4]`, and `0b00000100 = 4`, which belongs to `[0, 6]`.